src/tests/Common/XUnitLogChecker/XUnitLogChecker.cs

```
@@ -30,41 +30,116 @@ public TagResult(string matchValue, TagCategory matchCategory)
+        //
+        // In the Libraries tests however, the test logs are stored and created
+        // in a different way, currently outside the scope of the XUnitLogChecker,
+        // and therefore only the dump checking functionality is required in
+        // those cases, so we skip directly to that step.
+
+        if (s_configuration.HasResultsPath() && s_configuration.HasTestWrapper())
         {
-            WriteLineTimestamp("The path to the log file and"
-                               + " the name of the wrapper are required for an"
-                               + " accurate check and fixing.");
-            return MISSING_ARGS;
+            WriteLineTimestamp("The full run will be done.");
+            exitCode = DoFullRun();
         }
+        else
+        {
+            WriteLineTimestamp("Only dumps checking will be done.");
+            DoDumpsCheck(string.Empty);
+            exitCode = SUCCESS;
+        }
+
+        return exitCode;
+    }
 
-        // Creating variables for code clarity and ease of understanding.
+    private static void ParseCommandLineArgs(string[] args)
+    {
+        // The command-line arguments will always come in pairs of '--flag value'.
+        // Hence, we process two elements of the args array at a time.
 
-        string resultsDir = args[0];
-        string wrapperName = args[1];
+        for (int i = 0; i < args.Length; i+=2)
+        {
+            string nextOption = args[i];
+            string nextValue = (i+1) < args.Length ? args[i+1] : string.Empty;
+
+            switch (nextOption)
+            {
+                case "--results-path":
+                    s_configuration.ResultsPath = nextValue;
+                    break;
+
+                case "--test-wrapper":
+                    s_configuration.TestWrapperName = nextValue;
+                    break;
+
+                case "--dumps-path":
+                    s_configuration.DumpsPath = nextValue;
+                    break;
```

For the 3 cases, if `nextValue` is `string.Empty`, I think we should throw an exception indicating that "a value for <case name> was not provided". This is especially necessary since `DoFullRun`'s first few lines assume the TestWrapperName` and `ResultsPath` have a valid non-empty value.

---

We are already checking for the arguments to not be empty where needed so this is not necessary.

---

If a value isn't passed, the forloop is going to misbehave.

`XUnitLogChecker.exe --results-path --test-wrapper --dumps-path`

---

This is not meant to be a tool used for general consumption. This is specifically for our test infrastructure with specific calls that should be made correctly in the first place (i.e. that case you describe should never happen)